src/jslogger.js

```diff
@@ -0,0 +1,320 @@
+            }
+
+            appender(data);
+            return;
+        }
+
+        // store data to localStorage
+        function appender(data) {
+
+            count += 1;
+            window.localStorage.setItem("logging_" + data.UUID, JSON.stringify(data));
+            if(maxLogs == 0) {
+                return;
+            }
+
+            if(count > maxLogs) {
+                ajaxCall();     // flush logs to server side.
+                count = 0;
+            }
+            return;
+        }
+
+        // Data for logging functions
+        function log_data(level, msg, url, misc){
+            var data = {
+                type    : level,
+                message : msg,
+                url     : url,
+                misc    : misc
+            }
+            common(data);
+            return;
+        }
+
+
+        // Log different kinds of console logging methods
+        function info() {
+
+            log_data("info", arguments[0], window.location.href, arguments[1]);
+            console.info(arguments[0]);
+            return;
+        }
+
+        function error() {
+
+            log_data("exception", arguments[0], window.location.href, arguments[1]);
+            console.error(arguments[0]);
+            return;
+        }
+
+        function debug() {
+
+            log_data("debug", arguments[0], window.location.href, arguments[1]);
+            console.debug(arguments[0]);
+            return;
+        }
+
+        function log() {
+
+            log_data("log", arguments[0], window.location.href, arguments[1]);
+            console.log(arguments[0]);
+            return;
+        }
+
+        function warn() {
+
+            log_data("warn", arguments[0], window.location.href, arguments[1]);
+            console.warn(arguments[0]);
+            return;
+        }
+
+        function msg() {
+
+            log_data("msg", arguments[0], window.location.href, arguments[1]);
+            return;
+        }
+
+        function exception() {
+
+            log_data("exception", arguments[0], window.location.href, arguments[1]);
+            console.error(arguments[0]);
+            return;
+        }
+
+        function xhrStatus(msg, req_url, xhr, startTime, is_call_success) {
+
+            var endTime = new Date().getTime(),
+                data   = {
+                status          : xhr.status,
+                ready_state     : xhr.readyState,
+                status_message  : xhr.statusText,
+                url             : req_url,
+                start_time_ms   : startTime,
+                end_time_ms     : endTime,
+                request_time_ms : endTime-startTime,
+                response_length : xhr.responseText.length
+            }
+
+            if(!is_call_success) {
+                data.response = xhr.responseText.substring(0,2000);
+            }
+
+            common({message:msg, misc:data});
+            return;
+        }
+
+        // Log ajax call
+        function ajax(data) {
```

Comment:
Do not mix logger and AJAX calls, write your AJAX middleware as a separate module and use logger there

Reply:
@prrashi Totally agree. @jaswanth098 This isn't high priority yet but is good to track so we can address later. Please create an issue in this repo for this.

Reply:
sure @prashanthellina will create a issue and track this over there.